lua/heirline/statusline.lua

```diff
@@ -272,32 +272,56 @@ local function register_global_function(component)
     return "v:lua." .. func_name
 end
 
+---@param data table
+local function create_event(data)
+    local event = {}
```

Comment:
The code would be clearer to me if you pluralized this variable as `events` since it's an array that can potentially contain multiple events.

Same with the name of the key in the table returned below:
```lua
    return {
        events = events,
        callback = data.callback,
        pattern = data.pattern,
    }
```